src/ts_catalog/compression_settings.c

```diff
@@ -177,37 +224,66 @@ ts_compression_settings_delete(Oid relid)
 	return count > 0;
 }
 
-TSDLLEXPORT void
-ts_compression_settings_rename_column_hypertable(Hypertable *ht, char *old, char *new)
+/*
+ * Delete entries matching the non-compressed relation.
+ */
+TSDLLEXPORT bool
+ts_compression_settings_delete(Oid relid)
 {
-	ts_compression_settings_rename_column(ht->main_table_relid, old, new);
-	if (ht->fd.compressed_hypertable_id)
-	{
-		ListCell *lc;
-		List *chunks = ts_chunk_get_by_hypertable_id(ht->fd.compressed_hypertable_id);
-		foreach (lc, chunks)
-		{
-			Chunk *chunk = lfirst(lc);
-			ts_compression_settings_rename_column(chunk->table_id, old, new);
-		}
-	}
+	return compression_settings_delete(relid, false);
 }
 
-TSDLLEXPORT void
-ts_compression_settings_rename_column(Oid relid, char *old, char *new)
+/*
+ * Delete entries matching a compressed relation.
+ */
+TSDLLEXPORT bool
+ts_compression_settings_delete_by_compress_relid(Oid relid)
 {
-	CompressionSettings *settings = ts_compression_settings_get(relid);
+	return compression_settings_delete(relid, true);
+}
 
-	if (!settings)
-		return;
+/*
+ * Delete entries matching either the primary key (non-compressed relation) or
+ * the secondary key (compressed relation).
+ */
+TSDLLEXPORT bool
+ts_compression_settings_delete_any(Oid relid)
+{
+	if (!ts_compression_settings_delete(relid))
+		return ts_compression_settings_delete_by_compress_relid(relid);
+	return true;
+}
 
+static void
+compression_settings_rename_column(CompressionSettings *settings, const char *old, const char *new)
+{
 	settings->fd.segmentby = ts_array_replace_text(settings->fd.segmentby, old, new);
-
 	settings->fd.orderby = ts_array_replace_text(settings->fd.orderby, old, new);
-
 	ts_compression_settings_update(settings);
 }
 
+TSDLLEXPORT void
+ts_compression_settings_rename_column_cascade(Oid parent_relid, const char *old, const char *new)
+{
+	CompressionSettings *settings = ts_compression_settings_get(parent_relid);
+
+	if (settings)
+		compression_settings_rename_column(settings, old, new);
+
+	List *children = find_inheritance_children(parent_relid, NoLock);
+	ListCell *lc;
+
+	foreach (lc, children)
+	{
+		Oid relid = lfirst_oid(lc);
+
+		settings = ts_compression_settings_get(relid);
+
+		if (settings)
+			compression_settings_rename_column(settings, old, new);
+	}
+}
```

Comment:
Just an observation: This looks like we are using named columns rather than attribute number to reference columns, which requires us to have these kinds of functions. It would be better to just use the attribute number directly in all our tables, but I do not think this is something that should be done as part of this PR.

It is essentially an additional indirect lookup layer for the attributes, just like the one we currently have for our catalogs because we have our own chunk and hypertable identifiers.